### PR TITLE
[Merged by Bors] - feat(Data.Fin): Add lemma `Fin.sub_succ_le_sub_of_le`

### PR DESCRIPTION
**Pull request by IvanRenison**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Fin/Basic.lean [hunk 1]
```diff
@@ -1484,6 +1484,13 @@ lemma pos_of_ne_zero {n : ℕ} {a : Fin (n + 1)} (h : a ≠ 0) :
     0 < a :=
   Nat.pos_of_ne_zero (val_ne_of_ne h)
 
+lemma sub_succ_le_sub_of_le {n : ℕ} {u v : Fin (n + 2)} (h : u < v) : v - (u + 1) < v - u  := by
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Fin/Basic.lean:1487]:
```suggestion
lemma sub_succ_le_sub_of_le {n : ℕ} {u v : Fin (n + 2)} (h : u < v) : v - (u + 1) < v - u := by
```

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

Thanks!

**Comment by semorrison** [comment]
@IvanRenison, apologies to hijack your PR, but while reviewing your proof I was sad that we didn't have better automation for this, so went away and wrote some!

I've pushed a commit that adds a `fin_omega` macro, which is just a preprocessor for the `omega` tactic, that then handles your new lemma as just `by fin_omega`.

If it is inappropriate for me to hijack your PR, please just let me know and we can merge as is, and then I'll do this in a separate PR. But if you don't object it seems less overhead to do it directly here.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**